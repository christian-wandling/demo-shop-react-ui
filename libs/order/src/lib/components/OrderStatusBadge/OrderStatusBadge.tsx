import React from 'react';
import { OrderStatusProps } from '../../models/orderStatusProps';

/**
 * Component for displaying the status of an order.
 *
 * This component displays the order status using different background colors
 * based on the status value. It renders the status text in a styled container with
 * conditional formatting:
 * - Green background for "Completed" status
 * - Orange background for "Created" status
 *
 * @example
 * <OrderStatus status={order.status} />
 */
export const OrderStatusBadge: React.FC<OrderStatusProps> = ({
  status,
  className = '',
  'data-testid': dataTestId = 'order-status',
}: OrderStatusProps) => {
  const getBackgroundClass = () => {
    switch (status) {
      case 'Completed':
        return 'bg-green-800';
      case 'Created':
        return 'bg-orange-500';
      default:
        return '';
    }
  };

  const formattedStatus = status.charAt(0).toUpperCase() + status.slice(1).toLowerCase();

  return (
    <div
      className={`${className} rounded-lg px-2 sm:px-4 py-1 sm:py-1.5 max-w-20 sm:max-w-28 text-center text-white border border-slate-300 text-xs ${getBackgroundClass()}`}
      data-testid={dataTestId}>
      {formattedStatus}
    </div>
  );
};
