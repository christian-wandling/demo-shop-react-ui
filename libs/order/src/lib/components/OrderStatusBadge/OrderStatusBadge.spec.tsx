import { describe, expect, it } from 'vitest';
import { render, screen } from '@testing-library/react';
import { OrderStatusBadge } from './OrderStatusBadge';

describe('OrderStatusBadge', () => {
  it('renders with Completed status and correct styling', () => {
    render(<OrderStatusBadge status="Completed" />);

    const el = screen.getByTestId('orderStatus');

    expect(el).toBeTruthy();
    expect(el.classList).toContain('bg-green-800');
    expect(el.classList).toContain('rounded-lg');
    expect(el.classList).toContain('text-white');
  });

  it('should match snapshot', () => {
    const { baseElement } = render(<OrderStatusBadge status="Completed" />);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders with Created status and correct styling', () => {
    render(<OrderStatusBadge status="Created" />);

    const el = screen.getByTestId('orderStatus');

    expect(el).toBeTruthy();
    expect(el.classList).toContain('bg-orange-500');
    expect(el.classList).toContain('rounded-lg');
    expect(el.classList).toContain('text-white');
  });

  it('properly formats status text with first letter capitalized', () => {
    render(<OrderStatusBadge status="Created" />);

    const el = screen.getByTestId('orderStatus');

    expect(el).toBeTruthy();
    expect(el.textContent).toMatch(/^[A-Z][a-z]*(\s[A-Z][a-z]*)*/);
  });

  it('applies responsive classes for padding and width', () => {
    render(<OrderStatusBadge status="Completed" />);

    const el = screen.getByTestId('orderStatus');

    expect(el.classList).toContain('px-2');
    expect(el.classList).toContain('sm:px-4');
    expect(el.classList).toContain('py-1');
    expect(el.classList).toContain('sm:py-1.5');
    expect(el.classList).toContain('max-w-20');
    expect(el.classList).toContain('sm:max-w-28');
  });
});
