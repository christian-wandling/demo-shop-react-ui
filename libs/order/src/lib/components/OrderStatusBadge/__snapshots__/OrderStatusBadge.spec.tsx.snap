// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OrderStatusBadge > should match snapshot 1`] = `
<body>
  <div>
    <div
      class=" rounded-lg px-2 sm:px-4 py-1 sm:py-1.5 max-w-20 sm:max-w-28 text-center text-white border border-slate-300 text-xs bg-green-800"
      data-testid="orderStatus"
    >
      Completed
    </div>
  </div>
</body>
`;
